client/js/jsonedit.js

```diff
@@ -830,6 +881,23 @@ function jeShowCommands() {
   if(jeSecondaryWidget)
     commandText += `\n\n${jeSecondaryWidget}\n`;
   $('#jeCommands').innerHTML = commandText;
+  var buttons = [... document.querySelectorAll('#jeCommands>button')];
+  if (buttons.length > 0) {
+    buttons.forEach((e) => {
+      if (e != undefined) document.getElementById(e.id).addEventListener('click', clickButton);
+    });
+  }
+}
```

Comment:
This looks way too complicated to me. No need to check for `length > 0` (`forEach` should be fine). No idea why you would expect `undefined` values inside the callback. And `e` already is a DOM element. So this should be the same:

```Javascript
let buttons = [... document.querySelectorAll('#jeCommands>button')];
buttons.forEach(e => e.addEventListener('click', clickButton));
```

There is actually a helper function `$a` for `document.querySelectorAll` though:

```Javascript
for(const e of $a('#jeCommands>button'))
  e.addEventListener('click', clickButton);
```

But there's even a helper function for that:

```Javascript
on('#jeCommands>button', 'click', clickButton);
```

_(Didn't test any of these.)_

Reply:
Yes, the `on` is much easier and worked perfectly.